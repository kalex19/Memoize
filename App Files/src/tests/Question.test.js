import React from 'react';
import Question from '../QuestionCard/QuestionCard.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const mockDataObj=   {
    "questionSetKey": 1,
    "language": "HTML",
    "codeSnippetQ": "https://imgur.com/9AC4cch",
    "answer": ["className should be class", "className should be class and the regexp[a - Z]\\w+ should be[A - Z]\\w+", "The label should be in a seperate div from the input div"],
    "answerId": 2
  };
const mockScore=0;
const mockQuestionSetKey=1;
const mockScoreAnimation = jest.fn();
const mockNoScoreAnimation=jest.fn();
const mockIncrementKey=jest.fn();
const mockIncrementScore=jest.fn();
const mockSwitchAnswer=jest.fn();
const mockEndGame=jest.fn();

describe('QuestionCard', () => {
    let wrapper;
  
    beforeEach(() => {
      wrapper = shallow(
        <QuestionCard
        dataObj={mockDataObj}
        score={mockScore}
        questionSetKey={mockQuestionSetKey}
        scoreAnimation ={mockScoreAnimation}
        noScoreAnimation={mockNoScoreAnimation}
        incrementKey={mockIncrementKey}
        incrementScore={mockIncrementScore}
        switchAnswer={mockSwitchAnswer}
        endGame={mockEndGame}
        />
      );
    });

    it('should match the snapshot with all data passed in', () => {
        expect(wrapper).toMatchSnapshot();
    });

      it('should have the proper default state', () => {
        expect(wrapper.state()).toEqual({codeSnippetQ: '' });
      });

      it('should increment the key when switching the answer ', () => {
         expect(mockIncrementKey).toHaveBeenCalled();
         expect(wrapper.state().topEqual({questionSetKey: 30}));
         expect(mockEndGame).toHaveBeenCalled();
      });

      it('should reset state of start to false ', () => {
        expect(mockEndGame).toHaveBeenCalled();
        expect(wrapper.state()).toEqual({start: false });
     });

  });