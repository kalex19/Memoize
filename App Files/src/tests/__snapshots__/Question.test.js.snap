// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard should match the snapshot with all data passed in 1`] = `undefined`;
